chess/ChessGameModel.swift

```diff
@@ -15,22 +16,88 @@ struct ChessGameModel {
     }
 
     private(set) var board: [[ChessPiece?]] = .init(repeating: .init(repeating: nil, count: 8), count: 8)
+    private let logger = Logger(subsystem: "com.khaletska.chess", category: "GameModel")
 
     mutating func createNewGameBoard(configuration: BoardConfiguration) {
         self.board = configuration.generateBoard()
     }
 
-    mutating func move(from: Coordinate, to: Coordinate) {
-        let piece = self.board[from.row][from.col]
-        self.board[to.row][to.col] = piece
-        self.board[from.row][from.col] = nil
+    mutating func pieceMoved(from source: Coordinate, to destination: Coordinate) throws {
```

Comment:
this method is an entry point to action handling

Reply:
pieceMoved calls movePiece 🫠

Reply:
what concerns you here? The idea is to separate intention and an actual action. We can use some other semantics instead of intention/action. Like calling public method `movePiece` will cause eventually to call  `shiftPiece` in response. Wdyt?

Reply:
```suggestion
    mutating func intentionToMovePiece(from source: Coordinate, to destination: Coordinate) throws {
```